src/TestFramework/TestFramework/Assertions/CollectionAssert.cs

```diff
@@ -603,27 +701,32 @@ public static void IsNotSubsetOf([NotNull] ICollection? subset, [NotNull] IColle
             return;
         }
 
+        DebugEx.Assert(actual is not null, "actual is not null here");
+
+        int expectedCollectionCount = expected.Count();
+        int actualCollectionCount = actual.Count();
+
         // Check whether the element counts are different.
-        if (expected.Count != actual!.Count)
+        if (expectedCollectionCount != actualCollectionCount)
         {
             string userMessage = Assert.BuildUserMessage(message, parameters);
             string finalMessage = string.Format(
                 CultureInfo.CurrentCulture,
                 FrameworkMessages.ElementNumbersDontMatch,
                 userMessage,
-                expected.Count,
-                actual.Count);
+                expectedCollectionCount,
+                actualCollectionCount);
             Assert.ThrowAssertFailed("CollectionAssert.AreEquivalent", finalMessage);
         }
 
         // If both collections are empty, they are equivalent.
-        if (expected.Count == 0)
+        if (!expected.Any())
         {
             return;
         }
 
         // Search for a mismatched element.
-        if (FindMismatchedElement(expected, actual, out int expectedCount, out int actualCount, out object? mismatchedElement))
+        if (FindMismatchedElement<T>(expected, actual, comparer, out int expectedCount, out int actualCount, out object? mismatchedElement))
```

Comment:
You probably don't need the explicit `T`

```suggestion
        if (FindMismatchedElement(expected, actual, comparer, out int expectedCount, out int actualCount, out object? mismatchedElement))
```